src/services/environment-manager.js

```diff
@@ -2,59 +2,21 @@ const P = require('bluebird');
 const agent = require('superagent-promise')(require('superagent'), P);
 const authenticator = require('./authenticator');
 const EnvironmentSerializer = require('../serializers/environment');
-const EnvironmentDeserializer = require('../deserializers/environment');
-const JobDeserializer = require('../deserializers/job');
+const environmentDeserializer = require('../deserializers/environment');
 const DeploymentRequestSerializer = require('../serializers/deployment-request');
-const ProgressBar = require('progress');
-const { promisify } = require('util');
-
-const setTimeoutAsync = promisify(setTimeout);
+const JobStateChecker = require('../services/job-state-checker');
 
 function EnvironmentManager(config) {
-  let bar;
-  this.checkState = async (jobId, authToken) => {
-    try {
-      const jobResponse = await agent
-        .get(`${config.serverHost}/api/jobs/${jobId}`)
-        .set('Authorization', `Bearer ${authToken}`)
-        .set('forest-origin', 'Lumber')
-        .set('forest-project-id', config.projectId)
-        .then(response => new JobDeserializer.deserialize(response.body));
-
-      if (jobResponse
-        && jobResponse.state) {
-        let isBarComplete = false;
-        if (jobResponse.progress) {
-          bar.update(jobResponse.progress / 100);
-          isBarComplete = bar.complete;
-        }
-        if (jobResponse.state !== 'inactive'
-          && jobResponse.state !== 'active') {
-          if (jobResponse.state === 'complete' && !isBarComplete) {
-            bar.update(1);
-          }
-          return jobResponse.state !== 'failed';
-        }
-      }
-    } catch (e) {
-      return false;
-    }
-
-    await setTimeoutAsync(1000);
-    return this.checkState(jobId, authToken);
-  };
-
   this.listEnvironments = async () => {
     const authToken = authenticator.getAuthToken();
 
-    /* eslint new-cap: off */
     return agent
       .get(`${config.serverHost}/api/projects/${config.projectId}/environments`)
       .set('Authorization', `Bearer ${authToken}`)
       .set('forest-origin', 'Lumber')
       .set('forest-project-id', config.projectId)
       .send()
-      .then(response => new EnvironmentDeserializer.deserialize(response.body));
+      .then(response => environmentDeserializer.deserialize(response.body));
```

Comment:
I don't see any change in the `environmentDeserializer` implementation. Are there missing commits?

Reply:
It just make no sens to do `new EnvironmentDeserializer.deserialize`. First `environmentDeserializer` is an instance so it shouldn't be name with a capitalized first letter. Then `deserialize` is not a constructor, it doesn't use `this` internally and it is not named `Deserialize`.

Reply:
I could have put this in a separate PR, sorry.